analyses/focal-cn-file-preparation/05-define-most-focal-cn-units.Rmd

```diff
@@ -0,0 +1,293 @@
+# above to be called the majority status -- the decision
+# for a cutoff of 90% here was made to ensure that the status
+# is not only the majority status but it is also significantly
+# called more than the other status values in the region
+status_threshold <- 0.9
+
+# The fraction threshold for determining if enough of a region
+# (arm, cytoband, or gene) is callable to determine its status --
+# the decision for a cutoff of 50% here was made as it seems reasonable
+# to expect a region to be more than 50% callable for a dominant status
+# call to be made
+uncallable_threshold <- 0.5
+```
+
+## Set up
+
+### Libraries and functions
+
+```{r}
+library(tidyverse)
+```
+
+### Files and directories
+
+```{r}
+results_dir <- "results"
+```
+
+### Read in files
+
+Read in cytoband status file and format it for what we will need in this notebook. 
+
+```{r}
+# Read in the file with consensus CN status data and the UCSC cytoband data --
+# generated in `03-add-cytoband-status-consensus.Rmd`
+consensus_seg_cytoband_status_df <-
+  read_tsv(file.path("results", "consensus_seg_with_ucsc_cytoband_status.tsv.gz")) %>%
+  # Need this to not have `chr`
+  mutate(
+    chr = gsub("chr", "", chr),
+    cytoband = paste0(chr, cytoband)
+  ) %>%
+  select(
+    chromosome_arm,
+    # Distinguish this dominant status that is based on cytobands, from the status
+    dominant_cytoband_status = dominant_status,
+    cytoband,
+    Kids_First_Biospecimen_ID,
+    band_length,
+    gain_fraction,
+    loss_fraction,
+    callable_fraction
+  )
+```
+
+Read in the gene-level data. 
+
+```{r}
+# Read in the annotated gene level CN file
+consensus_seg_autosomes_df <-
+  read_tsv(file.path(results_dir, "consensus_seg_annotated_cn_autosomes.tsv.gz")) %>%
+  mutate(chromosome_arm = gsub("(p|q).*", "\\1", cytoband))
+
+# Rename "amplification" status calls to be "gain" for the purpose of this script
+consensus_seg_autosomes_df$status <-
+  gsub("amplification", "gain", consensus_seg_autosomes_df$status)
+```
+
+## Define most focal units
+
+### Determine chromosome arm status
+
+```{r}
+# Use our custom function to make the arm status calls
+consensus_seg_arm_status <- consensus_seg_cytoband_status_df %>%
+    # Group by biospecimen ID and region
+    group_by(Kids_First_Biospecimen_ID, chromosome_arm) %>%
+    # Summarize the weighted means for each status
+    summarize(
+      loss_fraction = weighted.mean(loss_fraction, band_length),
+      gain_fraction = weighted.mean(gain_fraction, band_length),
+      callable_fraction = weighted.mean(callable_fraction, band_length)
+    ) %>%
+    # Define dominant status based the weighted means meeting a status
+    # threshold
+    mutate(
+      dominant_arm_status = case_when(
+        callable_fraction < (1 - uncallable_threshold) ~ "uncallable",
+        loss_fraction > status_threshold ~ "loss",
+        gain_fraction > status_threshold ~ "gain",
+        loss_fraction + gain_fraction > status_threshold ~ "unstable",
```

Comment:
I wonder if in an example situation where one status is >> than the other whether unstable is the right call. For example, if `loss_fraction_arm` is 0.89  and `gain_fraction_arm`  is 0.01 whether this should be called as `unstable` but I guess we could be asking these kinds of questions all day and maybe they don't actually happen often if at all?

Reply:
Yes and also yes. I think this is a place where refinement could happen, but I don't know how much of a difference it will make in final conclusions. If we were to refine it, we would probablyu need to add another threshold for unstable, but I don't think we would want to call a region with 80% gain and 10%  loss as "neutral". In some sense, if all this does is catch a few things that fall just below threshold, that might be a good thing. Note that the consequences for the final table are the same: neutral and unstable are treated essentially the same way when looking at smaller regions/genes.